### ISSUE TITLE
Source Notion: Migrate to advancedAuth

### ISSUE THREAD

**Issue by lazebnyi** [airbytehq/airbyte#26242]

**Comment by darynaishchenko** [comment]
moved to on hold, until this issue https://github.com/airbytehq/airbyte/pull/26683 is resolved

**Comment by davydov-d** [comment]
still blocked by https://github.com/airbytehq/airbyte/pull/26683#issuecomment-1591128185
